contracts/aggregator/fantom/flashloan/main.sol

```diff
@@ -0,0 +1,183 @@
+        );
+        console.log("_finBals: ", instaLoanVariables_._finBals[0]);
+        validateFlashloan(instaLoanVariables_);
+
+        status = 1;
+        return true;
+    }
+
+    /**
+     * @dev Main function for flashloan for all routes. Calls the middle functions according to routes.
+     * @notice Main function for flashloan for all routes. Calls the middle functions according to routes.
+     * @param _tokens token addresses for flashloan.
+     * @param _amounts list of amounts for the corresponding assets.
+     * @param _route route for flashloan.
+     * @param _data extra data passed.
+    */
+    function flashLoan(	
+        address[] memory _tokens,	
+        uint256[] memory _amounts,
+        uint256 _route,
+        bytes calldata _data,
+        bytes calldata // kept for future use by instadapp. Currently not used anywhere.
+    ) external {
+        require(_tokens.length == _amounts.length, "array-lengths-not-same");
+
+        (_tokens, _amounts) = bubbleSort(_tokens, _amounts);
+        validateTokens(_tokens);
+
+        if (_route == 1) {
+            revert("this route is only for mainnet, polygon, and avalanche");
+        } else if (_route == 2) {
+            revert("this route is only for mainnet");
+        } else if (_route == 3) {
+            revert("this route is only for mainnet");
+        } else if (_route == 4) {
+            revert("this route is only for mainnet");
+        } else if (_route == 5) {
+            revert("this route is only for mainnet, polygon and arbitrum");
+        } else if (_route == 6) {
+            revert("this route is only for mainnet");
+        } else if (_route == 7) {
+            revert("this route is only for mainnet and polygon");
+        } else if (_route == 8) {
+            revert("this route is only for arbitrum, polygon and optimism");
+        } else if (_route == 9) {
+            spell(AAVE_IMPL, msg.data);
+        } else if (_route == 10){
+            (_tokens, _amounts) = bubbleSort(_tokens, _amounts);
+            validateTokens(_tokens);
+            routeFLA(msg.sender, _tokens, _amounts, _data);
+        } else {
+            revert("route-does-not-exist");
+        }
+
+        emit LogFlashloan(
+            msg.sender,
+            _route,
+            _tokens,
+            _amounts
+        );
+    }
+
+    /**
+     * @dev Function to get the list of available routes.
+     * @notice Function to get the list of available routes.
+    */
+    function getRoutes() public view returns (uint16[] memory routes_) {
+        console.log("Inside getRoutes main");
+        routes_ = new uint16[](2);
+        routes_[0] = 9;
+        routes_[1] = 10;
+    }
+
+    /**
+     * @dev Function to transfer fee to the treasury.
+     * @notice Function to transfer fee to the treasury.
+     * @param _tokens token addresses for transferring fee to treasury.
+    */
+    function transferFeeToTreasury(address[] memory _tokens) public {
```

Comment:
Need to change this function like [this](https://github.com/Instadapp/flashloan-aggregator/pull/65)